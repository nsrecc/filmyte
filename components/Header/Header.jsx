/* eslint-disable jsx-a11y/anchor-is-valid */
import React from 'react';
import Link from 'next/link'; // Wraps <a> elements
import { useThemeContext } from 'providers/ThemeProvider/ThemeProvider';
import { Sun, Moon } from 'phosphor-react';
import styles from './Header.module.scss';

const Header = () => {
  const { themeIsLight, toggleTheme } = useThemeContext();

  // Props for dark mode button icon
  const iconProps = {
    color: themeIsLight ? '#000000' : '#ffffff',
    size: '1.5em',
    weight: 'bold',
  };

  /* eslint-disable react/jsx-props-no-spreading */
  const themeIcon = themeIsLight ? <Moon {...iconProps} /> : <Sun {...iconProps} />;
  /* eslint-enable react/jsx-props-no-spreading */

  return (
    <header className={styles.header}>
      <div className={styles.headerInner}>
        <Link href="/">
          <a className={styles.headerLogoLink}>
            <span className={styles.filmyteLogo}>Filmyte</span>
          </a>
        </Link>
        <div className={styles.headerEnd}>
          <nav className={styles.navWrapper}>
            <ul className={styles.navList}>
              <li>
                <Link href="/search">
                  <a>Search</a>
                </Link>
              </li>
              <li>
                <Link href="/discover">
                  <a>Discover</a>
                </Link>
              </li>
              <li>
                <Link href="/about">
                  <a>About</a>
                </Link>
              </li>
            </ul>
          </nav>
          <button
            className={styles.themeButton}
            type="button"
            id="theme-toggle"
            onClick={toggleTheme}
          >
            {themeIcon}
          </button>
        </div>
      </div>
    </header>
  );
};

export default Header;
