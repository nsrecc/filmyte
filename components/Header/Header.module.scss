@use "styles/variables" as v;
@use "styles/mixins" as m;

.header {
  @include m.maxWidthWrapper;
  background-color: var(--theme-header-bg);

  .headerInner {
    @include m.maxWidth;
    padding: 1rem 2rem;
    display: inline-flex;
    justify-content: space-between;
    align-items: center;

    .headerLogoLink {
      padding: 0.5rem; // added padding to increase mouse click area
      
      .filmyteLogo {
        @include m.filmyteLogo;
        font-size: 2.2em;
      }
    }

    .headerEnd {
      display: inline-flex;
      justify-content: center;
      align-items: center;

      .navWrapper {
        .navList {
          display: inline-flex;
          justify-content: center;
          align-items: center;
          list-style: none;
          padding: 0;
          margin: 0;
    
          // nav list items
          li {
            margin-left: 3rem;
    
            a {
              padding: 0.5rem; // added padding to increase mouse click area
              font-size: 1.5em;
              font-weight: bold;
            }
          }
        }
      }

      .themeButton {
        margin-left: 3rem;
        border-color: var(--theme-page-text);
      }
    }
  }
}
