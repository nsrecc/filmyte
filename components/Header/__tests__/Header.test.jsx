import React from 'react';
import { shallow, mount, render } from 'enzyme';
import Link from 'next/link';
import Header from '../Header';

describe('Header', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
  });

  it('uses next/link components for navbar elements', () => {
    const wrapper = mount(<Header />);
    expect(wrapper.find(Link).exists()).toBe(true);
  });
});
