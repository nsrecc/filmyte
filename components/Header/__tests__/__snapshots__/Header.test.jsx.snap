// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
<header
  className="header"
>
  <div
    className="headerInner"
  >
    <Link
      href="/"
    >
      <a
        className="headerLogoLink"
      >
        <span
          className="filmyteLogo"
        >
          Filmyte
        </span>
      </a>
    </Link>
    <div
      className="headerEnd"
    >
      <nav
        className="navWrapper"
      >
        <ul
          className="navList"
        >
          <li>
            <Link
              href="/discover"
            >
              <a>
                Discover
              </a>
            </Link>
          </li>
          <li>
            <Link
              href="/about"
            >
              <a>
                About
              </a>
            </Link>
          </li>
        </ul>
      </nav>
      <button
        className="themeButton"
        id="theme-toggle"
        onClick={[MockFunction toggleTheme]}
        type="button"
      >
        <Moon
          color="#000000"
          size="1.5em"
          weight="bold"
        />
      </button>
    </div>
  </div>
</header>
`;
